#! /bin/sh /usr/share/dpatch/dpatch-run
## next-ia64.dpatch by Samuel Mimram <smimram@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Disable not-really-needed ia64 cast which broke with gcc 4.0.

@DPATCH@
diff -urNad coq-8.1~gamma~/kernel/byterun/coq_interp.c coq-8.1~gamma/kernel/byterun/coq_interp.c
--- coq-8.1~gamma~/kernel/byterun/coq_interp.c	2006-10-27 13:29:22.000000000 +0000
+++ coq-8.1~gamma/kernel/byterun/coq_interp.c	2006-11-26 18:37:23.000000000 +0000
@@ -44,11 +44,7 @@
 #  ifdef DEBUG
 #    define Next goto next_instr
 #  else
-#    ifdef __ia64__
-#      define Next goto *(void *)(coq_jumptbl_base + *((uint32 *) pc)++)
-#    else
-#      define Next goto *(void *)(coq_jumptbl_base + *pc++)
-#    endif
+#    define Next goto *(void *)(coq_jumptbl_base + *pc++)
 #  endif
 #else 
 #  define Instruct(name) case name:
